xla/backends/profiler/gpu/cupti_tracer.h

```diff
@@ -48,6 +72,207 @@ struct CuptiTracerOptions {
   bool sync_devices_before_stop = false;
   // Whether to enable NVTX tracking, we need this for TensorRT tracking.
   bool enable_nvtx_tracking = false;
+  // PM sampling configuration (defaults are 2khz rate, 100ms decode)
+  // Only read during creation of a PM sampling object, later changes have 
+  // no effect
+  struct PmSamplingConfig* pm_sampling_config;
+};
+
+// Container class for all CUPTI pm sampling infrastructure
+// - Configuration
+// - Enablement / disablement
+// - Buffer creation
+// - Worker thread creation / control
+// Decoding of counter data is done in PmSamplingDecodeThread class
+class PmSamplingDevice {
```

Comment:
Please follow the following class format: https://google.github.io/styleguide/cppguide.html#Class_Format

> The public section should be first, followed by the protected and finally the private section.

Reply:
See also https://google.github.io/styleguide/cppguide.html#Declaration_Order.